tokamak.py

```diff
@@ -21,29 +23,80 @@ def read_radial_build(file):
 
 def build_torus(major_radius,minor_radius,radial_build):
 
-	layers = list(radial_build.keys())
-	thicknesses = list(radial_build.values())
+	if 'Inboard' in radial_build:
+	
+		inboard_layers = list(radial_build['Inboard'].keys())
+		inboard_thicknesses = list(radial_build['Inboard'].values())
+		outboard_layers = list(radial_build['Outboard'].keys())
+		outboard_thicknesses = list(radial_build['Outboard'].values())
+
+		if np.sum(inboard_thicknesses) > major_radius:
+			raise(ValueError, "volume overlap at tokamak center.")
+
+		cubit.init([''])
+
+		cubit.cylinder(4*major_radius,major_radius,major_radius,major_radius)
+
+		cubit.torus(major_radius,minor_radius)
+		cubit.cmd("group 'mat:Vacuum' add vol 2")
+
+		inboard_radii = np.cumsum(inboard_thicknesses) + minor_radius
+		for radius in inboard_radii:
+			cubit.torus(major_radius,radius)
+
+		inboard_id = len(inboard_thicknesses) + 4
+		for id, layer_name in zip(range(inboard_id-2,2,-1), reversed(inboard_layers)):
+			cubit.cmd("subtract volume "+str(id - 1)+" from volume "+str(id)+" keep_tool")
+			cubit.cmd("intersect vol 1 "+str(inboard_id-1)+" keep")
+			cubit.cmd("group 'mat:"+ layer_name +"' add vol " + str(inboard_id))
+			cubit.cmd("delete vol "+str(inboard_id-1))
+			inboard_id += 2
+
+		outboard_radii = np.cumsum(outboard_thicknesses) + minor_radius
```

Comment:
There seems to be an opportunity for modularity here, with the O/B being nearly the same as the I/B